ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -143,7 +214,7 @@ If running in a thread, it can be cancelled via the isCancelled supplier (which
    */
   public TransactionSelectionResults buildTransactionListForBlock() {
     pendingTransactions.selectTransactions(
-        pendingTransaction -> evaluateTransaction(pendingTransaction));
+        pendingTransaction -> evaluateTransaction(pendingTransaction, false));
```

Comment:
why false here for `reportFutureNonceTransactionsAsInvalid`

Reply:
If I understood the code correctly, this method is selecting transactions from the transaction pool. If a TX has a nonce ahead account current nonce, it will be skipped until it can be included in a future block. So, the transaction is not invalid because it was not removed/dropped from the transaction pool and will likely be picked up latter on.